fedimint-client/src/transaction/sm.rs

```diff
@@ -154,44 +107,37 @@ impl IntoDynInstance for TxSubmissionStates {
     }
 }
 
-async fn trigger_created_submit(
-    tx: Transaction,
-    next_submission: SystemTime,
-    context: DynGlobalClientContext,
-) -> Result<TransactionId, String> {
-    fedimint_core::task::sleep(
-        next_submission
-            .duration_since(now())
-            .unwrap_or(Duration::ZERO),
-    )
-    .await;
-
-    // TODO: get rid of state machine created->created loop and only rely on this
-    // loop
+async fn trigger_created_rejected(tx: Transaction, context: DynGlobalClientContext) {
     loop {
         match context.api().submit_transaction(tx.clone()).await {
-            Err(e) if e.is_retryable() => {
-                debug!("Got {e} while submitting transaction, will sleep for {RESUBMISSION_INTERVAL:?}");
-                sleep(RESUBMISSION_INTERVAL).await;
+            Ok(submission_result) => {
+                if let SubmissionResult::Rejected(..) = submission_result {
+                    return;
+                }
+            }
+            Err(error) => {
+                if !error.is_retryable() {
+                    warn!(target: LOG_TARGET, ?error, "Federation returned non-retryable error");
+                }
```

Comment:
We need an upper bound on re-trying (or at least back off exponentially, although after a restart that would need to be preserved to be effective) as otherwise state machines might accumulate and crash the entire client (maximum slots for requests, subscriptions aren't super reliable under high load, …).